amy/dashboard/views.py

```diff
@@ -107,12 +109,25 @@ def admin_dashboard(request):
 
 @login_required
 def trainee_dashboard(request):
-    # Workshops person taught at
-    workshops = request.user.task_set.select_related("role", "event")
+    qs = Person.objects.select_related("airport").prefetch_related(
+        "badges",
+        "lessons",
+        "domains",
+        "languages",
+        Prefetch(
+            "task_set",
+            queryset=Task.objects.select_related(),
```

Comment:
So I do not think it's generally a good idea to call select related without arguments. From the docs:

>There may be some situations where you wish to call select_related() with a lot of related objects, or where you don’t know all of the relations. In these cases it is possible to call select_related() with no arguments. This will follow all non-null foreign keys it can find - nullable foreign keys must be specified. This is not recommended in most cases as it is likely to make the underlying query more complex, and return more data, than is actually needed.

Reply:
Resolved